File: dan/Problems/Easy/String/686. Repeated String Match/solution.java
class Solution {
   
    
    //Most clear way
    public int repeatedStringMatch(String A, String B){
        String as = A;
        for (int cnt = 1; cnt <= (B.length()/A.length() + 2); cnt++, as += A)
            if (as.indexOf(B) != -1) return cnt;
        return -1;
    }
}

File: dan/Problems/Easy/String/686. Repeated String Match/solution.py
class Solution(object):
    def repeatedStringMatch(self, A, B):
        """
        :type A: str
        :type B: str
        :rtype: int
        """
        a, rep = A, 1
        for i in range(1, len(B)/len(A) + 3):
            if B in a:
                return i
            a += A
        return -1
        
